ui/js/controllers/perf/alerts.js

```diff
@@ -0,0 +1,149 @@
+    function AlertsCtrl($state, $stateParams, $scope, $rootScope, $http, $q,
+                        thUrl, ThRepositoryModel, ThOptionCollectionModel,
+                        ThResultSetModel, thDefaultRepo, PhSeries, PhAlerts,
+                        phTimeRanges, dateFilter, thDateFormat) {
+
+        $scope.alertSummaries = [];
+        $scope.getCappedMagnitude = function(percent) {
+            // arbitrary scale from 0-20% multiplied by 5, capped
+            // at 100 (so 20% regression == 100% bad)
+            return Math.min(Math.abs(percent)*5, 100);
+        };
+
+        function addAlertSummaries(alertSummaries) {
+            var resultSetToSummaryMap = {};
+
+            alertSummaries.forEach(function(alertSummary) {
+                if (!resultSetToSummaryMap[alertSummary.repository]) {
+                    resultSetToSummaryMap[alertSummary.repository] = {};
+                }
+                _.forEach([alertSummary.result_set_id, alertSummary.prev_result_set_id],
+                          function(resultSetId) {
+                              if (!resultSetToSummaryMap[alertSummary.repository][
+                                  resultSetId]) {
+                                  resultSetToSummaryMap[alertSummary.repository][
+                                      resultSetId] = [ alertSummary ];
+                              } else {
+                                  resultSetToSummaryMap[alertSummary.repository][
+                                      resultSetId].push(alertSummary);
+                              }
+                          });
+
+                alertSummary.alerts.forEach(function(alert) {
+                    alert.title = PhSeries.getSeriesName(
+                        alert, $scope.optionCollectionMap,
+                        {includePlatformInName: true});
+                });
+
+                alertSummary.title = alertSummary.repository;
+                if (alertSummary.alerts.length > 1) {
+                    alertSummary.title += " " +
+                        _.min(_.pluck(alertSummary.alerts, 'amount_pct')) + " - " +
+                        _.max(_.pluck(alertSummary.alerts, 'amount_pct')) + "%";
+                } else {
+                    alertSummary.title += " " + alertSummary.alerts[0].amount_pct + "%";
+                }
+                alertSummary.title += " " + _.uniq(
+                    _.map(alertSummary.alerts, function(a) {
+                        return PhSeries.getTestName(a, { abbreviate:true });
+                    })).sort().join(' / ');
+                alertSummary.platforms = _.uniq(_.pluck(alertSummary.alerts,
+                                                        'machine_platform')).sort();
+            });
+
+            $q.all(_.map(_.keys(resultSetToSummaryMap), function(repo) {
+                return ThResultSetModel.getResultSets(
+                    repo, 0, 1000,
```

Comment:
Yeah this is not entirely clear. The underlying model's API leads to a lack of clarity here. It's not hard to change so I'll just change it.